indexed_filesource.go

```diff
@@ -82,6 +86,9 @@ func (s *IndexedFileSource) SetLogger(l *zap.Logger) {
 }
 
 func (s *IndexedFileSource) run() error {
+	if s.cursor != nil && s.cursor.Step != StepNew && s.cursor.Step != StepIrreversible {
```

Comment:
Is this actually from another work stream?

Reply:
I found another issue with cursor behavior in IndexedFileSource, it was not honoring the LIB/BLOCK part of the cursor. Reworking the tests surfaced this issue, so I added the cursor handling. I'll add it to description.